src/universal_transfer_operator/data_providers/ftp/filesystem/ftp.py

```diff
@@ -0,0 +1,82 @@
+from typing import Any
+
+from airflow.providers.ftp.hooks.ftp import FTPHook
+
+from src.universal_transfer_operator.data_providers.base import DataProviders
+
+
+class FTPDataProvider(DataProviders, FTPHook):
+    """
+    FTPDataProvider class is responsible to create FTP connection, read dataset and write dataset.
+    It inherits FTPHook and implements DataProviders.
+    """
+
+    def read_dataset(
+        self,
+        remote_full_path: str,
+        local_full_path_or_buffer: str,
+        callback: Any = None,
+    ) -> Any:
```

Comment:
It would be great if we could be more specific about what this method returns. Would `None` be a good return value?